File: Sqrilizz-Reports/src/main/java/dev/sqrilizz/SQRILIZZREPORTS/ReportsTabCompleter.java
package dev.sqrilizz.SQRILIZZREPORTS;

import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

/**
 * TabCompleter для всех команд плагина Sqrilizz-Reports
 * Предоставляет автодополнение для команд и их аргументов
 */
public class ReportsTabCompleter implements TabCompleter {

    @Override
    public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) {
        List<String> completions = new ArrayList<>();
        
        switch (command.getName().toLowerCase()) {
            case "report":
                return handleReportCommand(sender, args);
            case "reports":
                return handleReportsCommand(sender, args);
            case "report-language":
                return handleLanguageCommand(sender, args);
            case "report-telegram":
                return handleTelegramCommand(sender, args);
            case "report-webhook":
                return handleWebhookCommand(sender, args);
            case "report-discord":
                return handleDiscordCommand(sender, args);
            case "report-stats":
            case "report-reload":
                return completions; // Нет аргументов
        }
        
        return completions;
    }
    
    /**
     * Автодополнение для команды /report
     */
    private List<String> handleReportCommand(CommandSender sender, String[] args) {
        List<String> completions = new ArrayList<>();
        
        if (args.length == 1) {
            // Первый аргумент - имя игрока
            String input = args[0].toLowerCase();
            for (Player player : Bukkit.getOnlinePlayers()) {
                if (player.getName().toLowerCase().startsWith(input) && !player.equals(sender)) {
                    completions.add(player.getName());
                }
            }
        } else if (args.length == 2) {
            // Второй аргумент - примеры причин
            String input = args[1].toLowerCase();
            List<String> reasons = Arrays.asList(
                "читерство", "хак", "флай", "спидхак", "килаура", "антинокбек",
                "cheating", "hacking", "fly", "speed", "killaura", "antikb",
                "غش", "هاك", "طيران", "سرعة"
            );
            
            for (String reason : reasons) {
                if (reason.toLowerCase().startsWith(input)) {
                    completions.add(reason);
                }
            }
        }
        
        return completions;
    }
    
    /**
     * Автодополнение для команды /reports
     */
    private List<String> handleReportsCommand(CommandSender sender, String[] args) {
        List<String> completions = new ArrayList<>();
        
        if (args.length == 1) {
            // Первый аргумент - подкоманды
            String input = args[0].toLowerCase();
            List<String> subcommands = Arrays.asList("check", "clear", "clearall", "false");
            
            for (String subcmd : subcommands) {
                if (subcmd.startsWith(input)) {
                    completions.add(subcmd);
                }
            }
        } else if (args.length == 2 && !args[0].equalsIgnoreCase("clearall")) {
            // Второй аргумент - имя игрока (кроме clearall)
            String input = args[1].toLowerCase();
            for (Player player : Bukkit.getOnlinePlayers()) {
                if (player.getName().toLowerCase().startsWith(input)) {
                    completions.add(player.getName());
                }
            }
            
            // Добавляем оффлайн игроков из reports.yml
            try {
                for (String playerName : ReportManager.getReports().keySet()) {
                    if (playerName.toLowerCase().startsWith(input)) {
                        completions.add(playerName);
                    }
                }
            } catch (Exception e) {
                // Игнорируем ошибки
            }
        }
        
        return completions;
    }
    
    /**
     * Автодополнение для команды /report-language
     */
    private List<String> handleLanguageCommand(CommandSender sender, String[] args) {
        List<String> completions = new ArrayList<>();
        
        if (args.length == 1) {
            String input = args[0].toLowerCase();
            List<String> languages = Arrays.asList("en", "ru", "ar");
            
            for (String lang : languages) {
                if (lang.startsWith(input)) {
                    completions.add(lang);
                }
            }
        }
        
        return completions;
    }
    
    /**
     * Автодополнение для команды /report-telegram
     */
    private List<String> handleTelegramCommand(CommandSender sender, String[] args) {
        List<String> completions = new ArrayList<>();
        
        if (args.length == 1) {
            String input = args[0].toLowerCase();
            List<String> options = Arrays.asList("token", "chat");
            
            for (String option : options) {
                if (option.startsWith(input)) {
                    completions.add(option);
                }
            }
        }
        
        return completions;
    }
    
    /**
     * Автодополнение для команды /report-webhook
     */
    private List<String> handleWebhookCommand(CommandSender sender, String[] args) {
        List<String> completions = new ArrayList<>();
        
        if (args.length == 1) {
            String input = args[0].toLowerCase();
            List<String> options = Arrays.asList("set", "remove");
            
            for (String option : options) {
                if (option.startsWith(input)) {
                    completions.add(option);
                }
            }
        } else if (args.length == 2 && args[0].equalsIgnoreCase("set")) {
            // Пример URL для webhook
            completions.add("https://discord.com/api/webhooks/...");
        }
        
        return completions;
    }
    
    /**
     * Автодополнение для команды /report-discord
     */
    private List<String> handleDiscordCommand(CommandSender sender, String[] args) {
        List<String> completions = new ArrayList<>();
        
        if (args.length == 1) {
            String input = args[0].toLowerCase();
            List<String> options = Arrays.asList("token", "guild", "channel", "enable", "disable", "moderation", "status");
            
            for (String option : options) {
                if (option.startsWith(input)) {
                    completions.add(option);
                }
            }
        } else if (args.length == 2) {
            String command = args[0].toLowerCase();
            if (command.equals("moderation")) {
                String input = args[1].toLowerCase();
                List<String> values = Arrays.asList("true", "false", "on", "off");
                
                for (String value : values) {
                    if (value.startsWith(input)) {
                        completions.add(value);
                    }
                }
            } else if (command.equals("enable") || command.equals("disable") || command.equals("status")) {
                // Эти команды не требуют второго аргумента
                return completions;
            }
        }
        
        return completions;
    }
}

File: Sqrilizz-Reports/src/main/kotlin/dev/sqrilizz/SQRILIZZREPORTS/ReportsTabCompleter.kt
package dev.sqrilizz.SQRILIZZREPORTS

import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.entity.Player

class ReportsTabCompleter : TabCompleter {

    override fun onTabComplete(sender: CommandSender, command: Command, alias: String, args: Array<out String>): List<String> =
        when (command.name.lowercase()) {
            "report" -> handleReportCommand(sender, args)
            "bugreport" -> handleBugReportCommand(sender, args)
            "reports" -> handleReportsCommand(sender, args)
            "report-language" -> handleLanguageCommand(args)
            "report-telegram" -> handleTelegramCommand(args)
            "report-webhook" -> handleWebhookCommand(args)
            "report-discord" -> handleDiscordCommand(args)
            else -> emptyList()
        }

    private fun handleReportCommand(sender: CommandSender, args: Array<out String>): List<String> {
        if (args.size == 1) {
            val input = args[0].lowercase()
            return Bukkit.getOnlinePlayers()
                .filter { it.name.lowercase().startsWith(input) && it != sender }
                .map { it.name }
        }
        if (args.size == 2) {
            val input = args[1].lowercase()
            val reasons = listOf(
                "читерство", "хак", "флай", "спидхак", "килаура", "антинокбек",
                "cheating", "hacking", "fly", "speed", "killaura", "antikb",
                "غش", "هاك", "طيران", "سرعة"
            )
            return reasons.filter { it.lowercase().startsWith(input) }
        }
        return emptyList()
    }

    private fun handleBugReportCommand(sender: CommandSender, args: Array<out String>): List<String> {
        if (args.size == 1) {
            val input = args[0].lowercase()
            val categories = listOf(
                "duplication", "dupe", "crash", "server-crash",
                "exploit", "glitch", "performance", "lag",
                "gameplay", "mechanic", "world", "generation",
                "inventory", "items", "commands", "cmd",
                "permissions", "perms", "economy", "money",
                "other", "misc"
            )
            return categories.filter { it.startsWith(input) }
        }
        if (args.size == 2) {
            val category = args[0].lowercase()
            val input = args[1].lowercase()
            val examples = when (category) {
                "duplication", "dupe" -> listOf("Items duplicating in chest", "Дюп предметов в сундуке", "مضاعفة العناصر")
                "crash", "server-crash" -> listOf("Server crashes when...", "Сервер крашится когда...", "تعطل السيرفر عند...")
                "exploit", "glitch" -> listOf("Can bypass protection", "Можно обойти защиту", "يمكن تجاوز الحماية")
                "performance", "lag" -> listOf("Severe lag in...", "Сильные лаги в...", "تأخير شديد في...")
                "gameplay", "mechanic" -> listOf("Broken mechanic...", "Сломанная механика...", "ميكانيكا معطلة...")
                "world", "generation" -> listOf("World generation bug", "Баг генерации мира", "خلل في توليد العالم")
                "inventory", "items" -> listOf("Items disappearing", "Предметы пропадают", "اختفاء العناصر")
                "commands", "cmd" -> listOf("Command not working", "Команда не работает", "الأمر لا يعمل")
                "permissions", "perms" -> listOf("Permission not applied", "Права не применяются", "الصلاحيات لا تعمل")
                "economy", "money" -> listOf("Money duplication", "Дюп денег", "مضاعفة الأموال")
                "other", "misc" -> listOf("Description of the issue", "Описание проблемы", "وصف المشكلة")
                else -> emptyList()
            }
            return examples.filter { it.lowercase().startsWith(input) }
        }
        return emptyList()
    }

    private fun handleReportsCommand(sender: CommandSender, args: Array<out String>): List<String> {
        if (args.size == 1) {
            val input = args[0].lowercase()
            return listOf("gui", "list", "clear", "check", "false", "clearall")
                .filter { it.startsWith(input) }
        }
        if (args.size == 2) {
            val subCommand = args[0].lowercase()
            val input = args[1].lowercase()
            if (subCommand in listOf("clear", "check", "false")) {
                return Bukkit.getOnlinePlayers()
                    .filter { it.name.lowercase().startsWith(input) }
                    .map { it.name }
            }
        }
        return emptyList()
    }

    private fun handleLanguageCommand(args: Array<out String>): List<String> {
        if (args.size == 1) {
            val input = args[0].lowercase()
            return listOf("en", "ru", "ar").filter { it.startsWith(input) }
        }
        return emptyList()
    }

    private fun handleTelegramCommand(args: Array<out String>): List<String> {
        if (args.size == 1) {
            val input = args[0].lowercase()
            return listOf("token", "chat").filter { it.startsWith(input) }
        }
        return emptyList()
    }

    private fun handleWebhookCommand(args: Array<out String>): List<String> {
        if (args.size == 1) {
            val input = args[0].lowercase()
            return listOf("set", "remove").filter { it.startsWith(input) }
        }
        return emptyList()
    }

    private fun handleDiscordCommand(args: Array<out String>): List<String> {
        if (args.size == 1) {
            val input = args[0].lowercase()
            return listOf("set", "remove", "test").filter { it.startsWith(input) }
        }
        return emptyList()
    }
}
